src/components/ReactorEditor/index.tsx

```diff
@@ -14,41 +14,41 @@ const transform: ValidReactor = (data: Type) => {
 }`
 
 interface Props {
+  name: string
   json: any | undefined
   onSuccess(data: JsWorkerSuccess): void
 }
 
-export function ReactorEditor({ json, onSuccess }: Props) {
-  const [editor, setEditor] =
-    useState<monaco.editor.IStandaloneCodeEditor | null>(null)
-  const monacoDOM = useRef<HTMLDivElement>(null)
-
+export function ReactorEditor({ name, json, onSuccess }: Props) {
   useUpdateMonacoTsTypes(json)
 
-  useEffect(() => {
-    if (monacoDOM) {
-      setEditor((editor) => {
-        if (editor) return editor
-
-        return monaco.editor.create(monacoDOM.current!, {
-          value: DEFAULT_CODE,
-          language: 'typescript',
-        })
-      })
-    }
-
-    return () => editor?.dispose()
-  }, [monacoDOM.current])
-
   const { jsError } = useProessJson({
+    editorName: name,
     json,
-    editor,
     onSuccess,
   })
 
+  /**
+   * Connect VDOM with monaco editor states
+   *
+   * @param monacoDOM Monaco editor DOM element
+   */
+  const handleContainerInitialized = useCallback(
+    (monacoDOM: HTMLDivElement | null) => {
+      // In react dev mode, this function may be called multiple times.
+      // To guard duplicated model and editor creation, I have to clean up properly.
+      if (monacoDOM) {
+        createEditor(name, monacoDOM, DEFAULT_CODE)
+      } else {
+        getEditor(name)?.dispose()
+      }
+    },
+    [name],
+  )
+
   return (
     <article>
-      <div className={cx('root')} ref={monacoDOM}></div>
+      <div key={name} className={cx('root')} ref={handleContainerInitialized} />
```

Comment:
The reason why `key` exists, is to simplify life cycle.

Theoretically, `name` props can be changed. When it changes, I have to deal editor reinitialization. But such complex state managing is hard to maintain. So I isolated using editor id (`name`) as react `key`. Thanks for this, I can ignore state transitions between different `name`